engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -70,11 +71,34 @@ case class FixedValuesValidator(possibleValues: List[FixedExpressionValue]) exte
   }
 }
 
-case object LiteralIntValidator extends ParameterValidator {
-
+case class RegExpParameterValidator(pattern: String, message: String, description: String) extends ParameterValidator {
   override def isValid(paramName: String, value: String, label: Option[String])
                       (implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
-
-    if (Try(value.toInt).isSuccess) valid(Unit) else invalid(ProcessCompilationError.InvalidLiteralIntValue(paramName, label, value))
+    //Blank value should be not validate - we want to chain validators
```

Comment:
I still do not understand this logic :/

Reply:
What will happend when we have `@Nullable @Literal` param? :)

Reply:
Good change. One more notice: `Pattern.compile(pattern)` should be done once - factory method + replaced String with Pattern in fields or at least `lazy val`. Compiling is expensive...

Reply:
Changed

Reply:
I still see `Pattern.compile` inside `valid` method

Reply:
Ok I've moved it.. But in this case it doesn't matter..